### ISSUE TITLE
Date picker TypeError: setAttribute is not a function setAttribute(scope.$parent, value);

### ISSUE THREAD

**Issue by FDiskas** [angular-ui/bootstrap#3155]
I got error then datepicker is closed after date is selected
Version: 0.12.0 - 2014-11-16
```javascript
TypeError: setAttribute is not a function
setAttribute(scope.$parent, value);
```

**Comment by antoinepairet** [comment]
Could you please create a plunkr showing the issue?
thx

**Comment by FDiskas** [comment]
http://plnkr.co/edit/lSMwdgXmXXu3VA2RQXeI?p=preview
Tried to make date picker on input focus. The error you can see in console.

**Comment by rvanbaalen** [comment]
This is still pending. For me, this error occurs when I try to set `datepicker-mode`:

```html
<input
        type="text"
        class="form-control"
        placeholder="{{ 'LABEL.PLEASE_CHOOSE' | translate }}"
        id="dob"
        name="dob"
        oh-validate-model-state
        ng-model="relation.dob"
        max-date="datepickerEighteen"
        datepicker-mode="year"
        datepicker-popup="dd-MM-yyyy"
        datepicker-options="dobOptions"
        is-open="dobOpened"
        required
        readonly="readonly">
```

The above example __does not throw the error__. That is because the `datepicker-mode` value is set to `year`. Datepicker does not read this as a valid value. Therefore I tried `'year'` as a value (with the single quotes).

That seems to work in terms of datepicker functionality: the datepicker opens with the year view.

```html
<input
        type="text"
        class="form-control"
        placeholder="{{ 'LABEL.PLEASE_CHOOSE' | translate }}"
        id="dob"
        name="dob"
        oh-validate-model-state
        ng-model="relation.dob"
        max-date="datepickerEighteen"
        datepicker-mode="'year'" <!-- mind the single quotes here ' ' -->
        datepicker-popup="dd-MM-yyyy"
        datepicker-options="dobOptions"
        is-open="dobOpened"
        required
        readonly="readonly">
```

But when I have `'year'` as a value, it throws the above mentioned error for the function called here [on line 494 in datepicker.js](https://github.com/angular-ui/bootstrap/blob/5f9afe5a86af0e207eaacd6a9b1f202fd78d7009/src/datepicker/datepicker.js#L494)

**Comment by trueqbit** [comment]
How big are the chances that the fix gets in? It always happens when datepicker-mode is specified as a string literal, whenever you switch the mode browsing the dates.

It's fixable by using a binding, though. So I assume that the correct fix would be to not let the scope watch the value at all, since it's a constant string anyway...

**Comment by jafin** [comment]
Another Plnkr for this, run in angular 1.3.15 and ui-bootstrap-tpls-0.13.0.js  http://plnkr.co/edit/QGlIlxgU4SfkQ7oWVOlX?p=preview

**Comment by rvanbaalen** [comment]
Fix landed in ed10899d6b4dc4ab3c46c6b0288ecd8d98cc852a

**Comment by john-jay** [comment]
The fix in 0.13.1 worked for me. Thanks! I am using "triple-quotes" for the mode:

input type="text" ng-model="DOB" datepicker-popup="MM-dd-yyyy" datepicker-mode="'year'" is-open="dobOpen" ng-click="dobOpen = true" />

**Comment by uberspeck** [comment]
I just updated to ui.bootstrap 1.1.0 and Angular 1.5 and I'm now seeing this issue when i have the mode set to `year`

**Comment by uberspeck** [comment]
Here's a [plnkr](http://plnkr.co/edit/1zOnEINQzVk1r8XUFoWw?p=preview). The error happens whenever click on anything within the popup...

**Comment by icfantv** [comment]
Reopening for investigation.

**Comment by zpisgod** [comment]
here's a workaround to avoid getting error:
a config option on $scope like this:
```
$scope.dateOptions = {
        datepickerMode: 'month',
        minMode: 'month'
    };
```
and then explicit the ‘min-mode', 'datepicker-mode' and 'datepicker-options' property in the html element from the config object defined on $scope like that
```
<input type="text" class="form-control" uib-datepicker-popup="dd/MM/yyyy" ng-model="today" is-open="calendarOpened" min-mode="dateOptions.minMode" datepicker-mode="dateOptions.datepickerMode" datepicker-options="dateOptions" ng-required="true" close-text="Close" />
```

**Comment by wesleycho** [comment]
This seems fixed in the 1.2 release - see [here](http://plnkr.co/edit/kg0xNQTLKTSn29YhriIV?p=preview) for an example.
